test/Mono.Linker.Tests/TestCasesRunner/LinkerTestLogger.cs

```diff
@@ -13,6 +13,9 @@ public struct MessageRecord
 
 		public void LogMessage (MessageContainer msBuildMessage)
 		{
+			if (msBuildMessage == MessageContainer.Empty)
```

Comment:
This only fixes the warning suppression for tests - but ConsoleLogger will still try to print out empty warnings. I think warning suppression should be handled outside of the loggers as it's a linker feature.